lighthouse-core/fraggle-rock/gather/driver.js

```diff
@@ -67,8 +72,10 @@ class Driver {
     if (this.defaultSession !== throwingSession) return;
     const status = {msg: 'Connecting to browser', id: 'lh:driver:connect'};
     log.time(status);
-    const session = await this._page.target().createCDPSession();
-    this.defaultSession = new ProtocolSession(session);
+    const cdpSession = await this._page.target().createCDPSession();
+    this._targetManager = new TargetManager(cdpSession);
+    await this._targetManager.enable();
+    this.defaultSession = this._targetManager.rootSession();
```

Comment:
Because `TargetManager` is in charge of creating FR `ProtocolSession`s from `CDPSession`s, and we want to eventually make those sessions available for gatherers to access directly, Driver now gets its `driver.defaultSession` from targetManager instead of making it itself. This is a little weird, but otherwise we have two different session objects pointing to the same thing. This seemed conceptually simpler (the set of all active sessions includes `defaultSession`) but open to ideas.

Reply:
Big fan of avoiding the double session object! (I was seeing that at the CDP traffic level. so weird.)